firebase-ai-ksp-processor/build.gradle.kts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+  kotlin("jvm")
+  id("java-library")
+  id("maven-publish")
+}
+
+dependencies {
+  testImplementation(kotlin("test"))
+  implementation(libs.symbol.processing.api)
+  implementation(libs.kotlinpoet.ksp)
+}
+
+tasks.test { useJUnitPlatform() }
+
+kotlin { jvmToolchain(21) }
```

Comment:
Is there a reason we're using java 21 here? The rest of our repo expects 17 at the moment